### DISCUSSION TITLE
Pagina web aspx e integrazione api

### DISCUSSION THREAD

**Discussion by zonahobby** [fattureincloud/api#52] [Q&A] [answered]
Buongiorno,
sto cercando di integrare le api su una pagina aspx (VB).
Con metodo GET funziona ma con il POST no!

Questo il codice funzionante (metodo GET)

        Dim token As String = "TOKEN"
        Dim api_url As String = "https://api-v2.fattureincloud.it/c/company_id/settings/payment_methods"

        Dim request As WebRequest = WebRequest.Create(api_url)
        request.Method = "GET"
        request.Headers("Authorization") = "Bearer " & token

        Dim response As WebResponse = request.GetResponse()

        Dim dataStream As Stream = response.GetResponseStream()
        Dim reader As New StreamReader(dataStream)
        Dim responseFromServer As String = reader.ReadToEnd()
        Dim result = responseFromServer
        reader.Close()
        response.Close()
        Return result
        
  Avete un esempio di codice con metodo POST?
  Grazie!!

**Comment by valmoz** [comment]
Buongiorno,
se possibile, le suggerirei di [integrare la SDK C# nel suo codice](https://github.com/fattureincloud/api/discussions/26).

In alternativa, le ripropongo [due esempi che ho postato su un'altra discussione in merito](https://github.com/fattureincloud/api/discussions/19#discussioncomment-3196236) (purtroppo non ho modo di testarli).

Ci faccia sapere se riuscirà a risolvere.

Grazie
Mauro

**Reply by valmoz** [reply]
La documentazione della C# SDK: https://developers.fattureincloud.it/docs/sdks/csharp-sdk 

Nella discussione che ho condiviso ho anche [tradotto la modalità di integrazione in italiano](https://github.com/fattureincloud/api/discussions/19#discussioncomment-3207307), nel caso in cui dovesse servirle.

**Comment by zonahobby** [comment, marked as answer]
L'ho risolta così:


        Dim token As String = "TOKEN"
        Dim api_url As String = "https://api-v2.fattureincloud.it/c/company_id/issued_documents"

        Dim request As WebRequest = WebRequest.Create(api_url)
        request.Method = "POST"
        request.Headers("Authorization") = "Bearer " & token

        System.Net.ServicePointManager.ServerCertificateValidationCallback =
        Function(se As Object,
        cert As System.Security.Cryptography.X509Certificates.X509Certificate,
        chain As System.Security.Cryptography.X509Certificates.X509Chain,
        sslerror As System.Net.Security.SslPolicyErrors) True

        Dim postdata As String
        postdata = codice_fatt
        Dim byteArray As Byte() = Encoding.UTF8.GetBytes(postdata)
        request.ContentType = "application/json"
        request.ContentLength = System.Text.Encoding.UTF8.GetBytes(postdata).Length
        Dim dataStream As Stream = request.GetRequestStream()
        dataStream.Write(byteArray, 0, byteArray.Length)
        dataStream.Close()
        Dim response As WebResponse = request.GetResponse()
        dataStream = response.GetResponseStream()
        Dim reader As New StreamReader(dataStream)
        Dim responseFromServer As String = reader.ReadToEnd()
        Dim risposta As String = responseFromServer

        Return risposta
        
        
        
        Grazie comunque a tutti!
